File: backend/src/bookmark/dto/quick-favorite.dto.ts
import { IsEnum, IsString, IsNotEmpty } from 'class-validator';
import { BookmarkType } from '@prisma/client';

export class QuickFavoriteDto {
  @IsEnum(BookmarkType)
  targetType: BookmarkType;

  @IsString()
  @IsNotEmpty()
  targetId: string;
}

File: backend-spring/src/main/java/com/ulticode/modules/bookmark/dto/QuickFavoriteDTO.java
package com.ulticode.modules.bookmark.dto;

import com.ulticode.modules.bookmark.entity.enums.BookmarkType;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

/**
 * DTO for quick favorite/unfavorite operation.
 */
@Data
public class QuickFavoriteDTO {
    @NotBlank(message = "Target ID is required")
    private String targetId;

    @NotNull(message = "Target type is required")
    private BookmarkType targetType;
}
